src/app/pages/categories/management/pages-categories-management.component.ts

```diff
@@ -118,9 +131,28 @@ export class PagesCategoriesManagementComponent implements OnInit, OnDestroy {
       if(categoryItem.id === category.id) {
         return category;
       }
-
+      
       return categoryItem;
     });
     this.filterCategories();
   }
+
+  private deleteCategory(): void {
+    this.loadingDialogService.show('Deleting category');
+
+    this.pagesTransactionCategoriesService.deleteTransactionCategory().subscribe({
+      next: () => {
+        this.initCategories();
```

Comment:
This is probably why after deleting category it is not removed from the page. It is good idea to approach this like this in case of having pagination for transactions but if we're not having it it is much faster to just remove category from the list than wait for backend response and replace whole list.